org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java

```diff
@@ -1234,5 +1235,45 @@ public void checkOppositeReferenceUsed(Assignment assignment) {
 			}
 		}
 	}
-
+	
+	@Check
+	public void checkAnnotations(AbstractRule rule) {
```

Comment:
@svenefftinge and I discussed offline that it would be OK for now to allow arbitrary annotations without requiring annotations declarations. In the future, if we choose to introduce annotations declarations, we can offer a smooth migration path by rising a warning (not an error!) for annotations without declarations.